Source/Core/DiscIO/Blob.h

```diff
@@ -74,6 +74,19 @@ class SectorReader : public IBlobReader
 	u64 m_cache_tags[CACHE_SIZE];
 };
 
+class CBlobBigEndianReader
+{
+public:
+	CBlobBigEndianReader(IBlobReader& reader) : m_reader(reader) {}
+
+	bool Read32(u64 offset, u32* buffer) const;
+	bool Read16(u64 offset, u16* buffer) const;
+	bool Read8(u64 offset, u8* buffer) const;
```

Comment:
Having 8/16/32 makes the size of the read more explicit in calls, so that looking up the type of the buffer variable isn't necessary. Having a template function would be nice if it meant that this can be used for reading an arbitrary amount of data, but because the function needs to call Swap16/Swap32 depending on the amount of data, it won't work.